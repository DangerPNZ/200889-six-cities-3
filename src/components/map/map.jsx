import React from 'react';
import PropTypes from 'prop-types';
import leaflet from 'leaflet';

const MapSetting = {
  CITY: [52.38333, 4.9],
  ICON: leaflet.icon({
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30]
  }),
  ZOOM: 12
};
export class Map extends React.PureComponent {
  constructor(props) {
    super(props);
    this.map = React.createRef();
  }
  componentDidMount() {
    this.mapInit();
  }
  render() {
    return (
      <div id="map" style={{height: `100%`}} ref={this.map}></div>
    );
  }
  mapInit() {
    // Данное условие нарушает концепцию Код не должен подстраиваться под тесты
    if (!this.map || !this.map.current) {
      return;
    }

    const map = leaflet.map(this.map.current, {
      center: MapSetting.CITY,
      zoom: MapSetting.ZOOM,
      zoomControl: false,
      marker: true
    });
    const icon = MapSetting.ICON;
    map.setView(MapSetting.CITY, MapSetting.ZOOM);
    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);
    this.props.offerCoords.map((offer) => {
      leaflet
      .marker(offer, {icon})
      .addTo(map);
    });
  }
}

Map.propTypes = {
  offerCoords: PropTypes.arrayOf(
      PropTypes.arrayOf(PropTypes.number.isRequired)
  ).isRequired
};
