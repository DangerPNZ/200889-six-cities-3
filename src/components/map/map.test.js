import React from 'react';
import renderer from 'react-test-renderer';
import {Map} from './map.jsx';

const TestDataValue = {
  COORDINATES: [
    [52.369553943508, 4.85309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3809553943508, 4.939309666406198]
  ]
};

it(`Map component structure test`, () => {
  const tree = renderer
  .create(
      <Map
        offerCoords = {TestDataValue.COORDINATES}
      />,
      {
        createNodeMock: () => document.createElement(`div`)
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
