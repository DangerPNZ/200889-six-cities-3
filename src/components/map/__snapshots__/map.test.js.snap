// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component structure test 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
